assets/src/stories-editor/blocks/amp-story-page-attachment/edit.js

```diff
@@ -21,56 +21,34 @@ import AttachmentContent from './attachment-content';
 /**
  * Edit component for the page attachment block rendered in the editor.
  */
-class PageAttachmentEdit extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			isOpen: false,
-		};
-
-		this.toggleAttachment = this.toggleAttachment.bind( this );
-	}
+const PageAttachmentEdit = ( {
+	attributes,
+	setAttributes,
+	backgroundColor,
+	customBackgroundColor,
+	textColor,
+} ) => {
+	const {
+		opacity,
+		openText,
+	} = attributes;
 
-	componentDidUpdate( prevProps ) {
-		const {
-			backgroundColor,
-			customBackgroundColor,
-			textColor,
-			setAttributes,
-		} = this.props;
+	const [ isOpen, setIsOpen ] = useState( false );
 
-		if (
-			backgroundColor !== prevProps.backgroundColor ||
-			customBackgroundColor !== prevProps.customBackgroundColor ||
-			textColor !== prevProps.textColor
-		) {
-			const { style, attachmentClass } = this.getWrapperAttributes();
-			const newAttributes = { wrapperStyle: style };
-			if ( textColor !== prevProps.textColor ) {
-				newAttributes.attachmentClass = attachmentClass;
-			}
-			setAttributes( newAttributes );
+	const toggleAttachment = ( open ) => {
+		if ( open !== isOpen ) {
```

Comment:
[`useState` only re-renders if the value has changed](https://reactjs.org/docs/hooks-reference.html#bailing-out-of-a-state-update), so this entire function can just be replaced with `setIsOpen`.